### ISSUE TITLE
`NapiClass` constructor cannot be subclassed

### ISSUE THREAD

**Issue by Jarred-Sumner** [oven-sh/bun#172]
Filing so I remember to fix this

This currently impacts `@resvg/resvg-js`

```
module.exports.Resvg = class Resvg extends _Resvg {
  constructor(svg, options) {
     super(svg, JSON.stringify(options))
```

The problem in Bun is this code:
```c++
   NapiClass* napi = jsDynamicCast<NapiClass*>(newTarget);
    if (UNLIKELY(!napi)) {
        JSC::throwVMError(globalObject, scope, JSC::createTypeError(globalObject, "NapiClass constructor called on an object that is not a NapiClass"_s));
        return JSC::JSValue::encode(JSC::jsUndefined());
    }

```

That `!napi` is being returned because `newTarget` is a subclass of NapiClass.

Maybe the prototype needs to be checked instead? I'm not sure.

**Comment by Jarred-Sumner** [comment]
I believe the fix here is, instead of `jsDynamicCast`, to check `inherits` and then do `jsCast`. I see that pattern in WebKit code a decent amount.

**Comment by yisibl** [comment]
I'm the author of [resvg-js](https://github.com/yisibl/resvg-js), thank you very much for bringing this to your attention. It would be great if this could be fixed before bun 1.0 is released.

**Comment by yisibl** [comment]
I tested bun v0.8.0 in macOS and it still doesn't work. @Jarred-Sumner Can you help and look at it again?

![image](https://github.com/oven-sh/bun/assets/2784308/6e7b25bd-e185-4da3-8220-216e1ae53688)

It won't run on a Mac M1 Pro or Mac Intel CPU.

**Comment by Jarred-Sumner** [comment]
Very strange.  This is the regression test we run in CI which tries to
catch this
https://github.com/oven-sh/bun/blob/main/test/js/third_party/resvg/bbox.test.js

**Comment by yisibl** [comment]
I submitted a test repository for reference: https://github.com/yisibl/resvg-js-bun-test

**Comment by Jarred-Sumner** [comment]
The cause is a transpiler bug introduced after the napi-related parts were fixed

Input:
```js
localFileExisted = existsSync(
  join(__dirname, "resvgjs.darwin-arm64.node")
);
try {
  localFileExisted
    ? (nativeBinding = require("./resvgjs.darwin-arm64.node"))
    : (nativeBinding = require("@resvg/resvg-js-darwin-arm64"));
} catch (e) {
  loadError = e;
}
```

Output:
```js
localFileExisted = existsSync(join(__dirname, "resvgjs.darwin-arm64.node"));
try {
  localFileExisted
    ? (nativeBinding = () => ({}))
    : (nativeBinding = require("/path/to/resvgjs.darwin-arm64.node"));
} catch (e) {
  loadError = e;
}
```

**Comment by yisibl** [comment]
Thanks for the quick find of the cause of the problem

**Comment by yisibl** [comment]
Thanks, I confirmed the fix in bun v0.8.1
